pkg/aws/client.go

```diff
@@ -304,33 +304,15 @@ func (c *awsClient) GetCreator() (*Creator, error) {
 
 // Checks if given credentials are valid.
 func (c *awsClient) ValidateCredentials() (bool, bool, error) {
-	callerIdentityOutput, err := c.stsClient.GetCallerIdentity(&sts.GetCallerIdentityInput{})
+	// Validate the AWS credentials by calling STS GetCallerIdentity
+	// This will fail if the AWS access key and secret key are invalid. This
+	// will also work for STS credentials with access key, secret key and session
+	// token
+	_, err := c.stsClient.GetCallerIdentity(&sts.GetCallerIdentityInput{})
 	if err != nil {
 		return false, false, err
 	}
 
-	// Get IAM credentials only to support a better error message
-	// to help a user identity which AWS credentials the AWS client
-	// is using
-	iamCredentials, err := c.GetIAMCredentials()
-	if err != nil {
-		return false, false, err
-	}
-
-	parsedArn, err := arn.Parse(*callerIdentityOutput.Arn)
```

Comment:
I wonder if we should retain this call and `return true, parsedArn.Service == "sts", nil` at the end of the function, since during `rosa init` we have some code to capture in grafana how many users are using STS credentials. Probably not as important now that STS support is a thing

Reply:
I managed to find a solution in the latest commit. Let me know what you think! Can do this too.